src/repr/src/scalar.rs

```diff
@@ -1290,6 +1292,29 @@ impl<'a, E> DatumType<'a, E> for RegType {
     }
 }
 
+impl AsColumnType for Int2Vector {
+    fn as_column_type() -> ColumnType {
+        ScalarType::Int2Vector.nullable(true)
```

Comment:
when I change this to `ScalarType::String` it will stop crashing, but the column type is wrong, and a subsequent cast to `int2[]` will fail.